featuretools/entityset/entity.py

```diff
@@ -404,16 +404,21 @@ def delete_variables(self, variable_ids):
     def set_time_index(self, variable_id, already_sorted=False):
         # check time type
         if isinstance(self.df, dd.core.DataFrame):
-            self.time_index = variable_id
-            self.entityset.time_type = self.variable_types[variable_id]
-            return
-
-        if len(self.df) == 0:
-            time_to_check = vtypes.DEFAULT_DTYPE_VALUES[self[variable_id]._default_pandas_dtype]
+            already_sorted = True  # skip sorting
+            old_vtype = self.variable_types[variable_id]
+            if old_vtype == vtypes.Datetime:
+                time_type = vtypes.DatetimeTimeIndex
+            elif old_vtype in (vtypes.Numeric, vtypes.Ordinal):
+                time_type = vtypes.NumericTimeIndex
+            else:
+                time_type = old_vtype
         else:
-            time_to_check = self.df[variable_id].head(1).iloc[0]
+            if len(self.df) == 0:
```

Comment:
I think we may be able to simplify further by changing this line to 

```python
if isinstance(self.df, dd.core.DataFrame) or len(self.df) == 0:
```
and removing the conditional logic about `old_vtype`

Reply:
If you do that, won't you always force a `compute()` to get the length if `self.df` is a Dask dataframe?

Reply:
I think if the first statement of the or is true, the second is not evaluated

https://docs.python.org/3/reference/expressions.html#boolean-operations

Reply:
Not sure if I'm understanding correctly, but we could get into trouble calling .head on a dask dataframe since it by default grabs the first partition which could be empty. Since we're requiring vtypes when creating an entity, using the vtype provided to figure out the index type seems like the safer route.

Reply:
I'm proposing the logic look like this

```python
        if isinstance(self.df, dd.core.DataFrame) or len(self.df) == 0:
            time_to_check = vtypes.DEFAULT_DTYPE_VALUES[self[variable_id]._default_pandas_dtype]
        else:
            time_to_check = self.df[variable_id].head(1).iloc[0]
```